src/status_im/ui/screens/chat/pinned_messages.cljs

```diff
@@ -0,0 +1,121 @@
+(ns status-im.ui.screens.chat.pinned-messages
+  (:require [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.i18n.i18n :as i18n]
+            [status-im.ui.components.connectivity.view :as connectivity]
+            [status-im.ui.components.react :as react]
+            [quo.animated :as animated]
+            [status-im.ui.screens.chat.message.message :as message]
+            [status-im.ui.screens.chat.styles.main :as style]
+            [status-im.ui.screens.chat.message.gap :as gap]
+            [status-im.ui.screens.chat.components.accessory :as accessory]
+            [status-im.ui.screens.chat.message.datemark :as message-datemark]
+            [status-im.utils.platform :as platform]
+            [quo.react :as quo.react]
+            [status-im.ui.components.topbar :as topbar]
+            [status-im.ui.screens.chat.views :as chat]
+            [status-im.ui.components.list.views :as list]))
+
+(defn pins-topbar []
+  (let [{:keys [group-chat chat-id chat-name]}
+        @(re-frame/subscribe [:chats/current-chat])
+        pinned-messages @(re-frame/subscribe [:chats/pinned chat-id])
+        [first-name _] (when-not group-chat @(re-frame.core/subscribe [:contacts/contact-two-names-by-identity chat-id]))]
+    [topbar/topbar {:show-border? true
+                    :title        (if group-chat chat-name first-name)
+                    :subtitle     (if (= (count pinned-messages) 0)
+                                    (i18n/label :t/no-pinned-messages)
+                                    (i18n/label-pluralize (count pinned-messages) :t/pinned-messages-count))}]))
+
+(defn get-space-keeper-ios [bottom-space panel-space active-panel text-input-ref]
+  (fn [state]
+    ;; NOTE: Only iOS now because we use soft input resize screen on android
+    (when platform/ios?
+      (cond
+        (and state
+             (< @bottom-space @panel-space)
+             (not @active-panel))
+        (reset! bottom-space @panel-space)
+
+        (and (not state)
+             (< @panel-space @bottom-space))
+        (do
+          (some-> ^js (quo.react/current-ref text-input-ref) .focus)
+          (reset! panel-space @bottom-space)
+          (reset! bottom-space 0))))))
+
+(defn pinned-messages-empty []
+  [react/view {:style {:flex 1
+                       :align-items :center
+                       :justify-content :center}}
+   [react/text {:style style/intro-header-description}
+    (i18n/label :t/pinned-messages-empty)]])
+
+(defonce messages-list-ref (atom nil))
+
+(def list-key-fn #(or (:message-id %) (:value %)))
+(def list-ref #(reset! messages-list-ref %))
+
+(defn render-pin-fn [{:keys [outgoing type] :as message}
```

Comment:
can't we use regular message rendering? how's it different?

Reply:
we are forcing pinned to true but it is probably redundant, i think it should work, i'll try it

Reply:
done!